src_erl/Communication_Layer/http_Nerlserver/src/Client/clientStatem.erl

```diff
@@ -343,15 +345,24 @@ http_request(Host, Port,Path, Body)->
   httpc:request(post,{URL, [],"application/x-www-form-urlencoded",Body}, [], []).
 
 
-ack(MyName, PortMap) ->
+ack(MyName, NerlnetGraph) ->
     io:format("~p sending ACK   ~n",[MyName]),
-  {RouterHost,RouterPort} = maps:get(mainServer,PortMap),
+  {RouterHost,RouterPort} = getShortPath(MyName,"mainServer",NerlnetGraph),
 %%  send an ACK to mainserver that the CSV file is ready
-  http_request(RouterHost,RouterPort,"clientReady",atom_to_list(MyName)).
+  http_request(RouterHost,RouterPort,"clientReady",MyName).
+
 
 
+getShortPath(From,To,NerlnetGraph) when is_atom(To)-> 
+  	First = lists:nth(2,digraph:get_short_path(NerlnetGraph,From,atom_to_list(To))),
```

Comment:
Why do you convert from atom to list?
list requires more comparisons for equality test.
Atoms in Erlang are treated as enums.